internal/compiler-bridge/src-2.10/main/scala/xsbt/DelegatingReporter.scala

```diff
@@ -5,18 +5,48 @@ package xsbt
 
 import xsbti.{ F0, Logger, Maybe }
 import java.io.File
+import java.util.Optional
```

Comment:
Why `java.util.Optional` over `Option`? I don't like to hypothesize and I think I see your inner thoughts in this decision, but is it really worth it to clutter Scala's API with `Optional` for binary compatibility? Or is it because the consumer is a Java source? I haven't seen Java sources in this PR so that's why I'm discarding this option to make my point.
  
I wonder whether it would be useful to ask the Compiler team that they promise binary compatibility *forever* in easy-to-implement and recurrent classes like `Option`.